src/commonMain/kotlin/fr/acinq/eclair/db/PaymentsDb.kt

```diff
@@ -128,15 +131,15 @@ data class IncomingPayment(val preimage: ByteVector32, val origin: Origin, val r
     data class Received(val amount: MilliSatoshi, val receivedWith: ReceivedWith, val receivedAt: Long = currentTimestampMillis())
 
     sealed class ReceivedWith {
-        abstract val fees: MilliSatoshi
+        abstract val fees: MilliSatoshi?
 
         /** Payment was received via existing lightning channels. */
         object LightningPayment : ReceivedWith() {
-            override val fees: MilliSatoshi = 0.msat
+            override val fees: MilliSatoshi? = null // with Lightning, the fee is paid by the sender
```

Comment:
I'd rather make it a fee of 0 sat than a null, what do you think?

Reply:
I wanted to make it explicit that this fee is optional, not just = 0 sometimes. This is mostly useful for swap-ins where the peer may not send the `ChannelOrigin` object containing the fee, hence the `ReceivedWith` fee attribute becomes nullable.

And so I also updated the fee for incoming lightning payments where having a fee does not make sense.

Reply:
The issue I have with nullable here is that we don't know if it means:

1. There was no fee for this operation (it's free)
2. We're missing the information about the fee (but there could be one)

Whereas explicitly setting it to 0 sat makes it obvious where we say "this operation doesn't cost any fee" and we can easily tell whether the code is correct or if we're in fact not transmitting enough information to compute the fee correctly.

But if you still think nullable makes more sense it's your call!

Reply:
True, we would need an "unknown" option and a "do not apply" option to be really explicit. So let's keep it simple and use 0 msat instead.

Reply:
done in 419e904